src/settings.cpp

```diff
@@ -1550,25 +1600,58 @@ static void GRFSaveConfig(IniFile *ini, const char *grpname, const GRFConfig *li
 }
 
 /* Common handler for saving/loading variables to the configuration file */
-static void HandleSettingDescs(IniFile *ini, SettingDescProc *proc, SettingDescProcList *proc_list, bool only_startup = false)
+static void HandleSettingDescs(IniFile *generic_ini, IniFile *private_ini, IniFile *secrets_ini, SettingDescProc *proc, SettingDescProcList *proc_list, bool only_startup = false)
 {
-	proc(ini, _misc_settings,    "misc",  nullptr, only_startup);
+	proc(generic_ini, _misc_settings, "misc", nullptr, only_startup);
 #if defined(_WIN32) && !defined(DEDICATED)
-	proc(ini, _win32_settings,   "win32", nullptr, only_startup);
+	proc(generic_ini, _win32_settings, "win32", nullptr, only_startup);
 #endif /* _WIN32 */
 
-	for (auto &table : _setting_tables) {
-		/* The name "patch" is a fallback, as every setting should sets its own group. */
-		proc(ini, table, "patches", &_settings_newgame, only_startup);
+	/* The name "patches" is a fallback, as every setting should sets its own group. */
+
+	for (auto &table : _generic_setting_tables) {
+		proc(generic_ini, table, "patches", &_settings_newgame, only_startup);
+	}
+	for (auto &table : _private_setting_tables) {
+		proc(private_ini, table, "patches", &_settings_newgame, only_startup);
+	}
+	for (auto &table : _secrets_setting_tables) {
+		proc(secrets_ini, table, "patches", &_settings_newgame, only_startup);
 	}
 
-	proc(ini, _currency_settings,"currency", &_custom_currency, only_startup);
-	proc(ini, _company_settings, "company",  &_settings_client.company, only_startup);
+	proc(generic_ini, _currency_settings,"currency", &_custom_currency, only_startup);
```

Comment:
Existing missing space between _currency_settings, and "currency". Though if it gets touched, maybe fix that as well?